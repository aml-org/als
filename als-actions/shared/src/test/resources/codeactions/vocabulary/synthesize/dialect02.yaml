#%Dialect 1.0
dialect: Basic
version: 2
uses:
  customVoc: vocabulary.yaml

documents:
  root:
    encodes: RootNode
nodeMappings:
  A:
    mapping:
      text:
        propertyTerm: customVoc.propTerm
        range: string
  B:
    mapping:
      text:
        range: number
      newProp:
        range: A
  S3ResourceNode:
    classTerm: customVoc.classT
    mapping:
      IDs:
        sorted: false

        allowMultiple: false
        range: string
  RootNode:
    classTerm: v.Union
    union:
      - A
      - B
    typeDiscriminatorName: kind
    typeDiscriminator:
      TypeA: A
      TypeB: B