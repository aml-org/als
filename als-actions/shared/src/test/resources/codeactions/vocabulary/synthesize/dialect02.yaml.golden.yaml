documentChanges:
  file://als-actions/shared/src/test/resources/codeactions/vocabulary/synthesize/vocabulary1.yaml: create
  file://als-actions/shared/src/test/resources/codeactions/vocabulary/synthesize/vocabulary1.yaml:
    -
      from:
        line: 0
        column: 0
      to:
        line: 0
        column: 0
      content: |
        +
        #%Vocabulary 1.0
        base: http://a.ml/vocabulary/#
        vocabulary: vocabulary1
        classTerms:
          A:
            displayName: A
          B:
            displayName: B
        propertyTerms:
          text:
            displayName: text
            range: number
          newProp:
            displayName: newProp
            range: A
          iDs:
            displayName: IDs
            range: string
  file://als-actions/shared/src/test/resources/codeactions/vocabulary/synthesize/dialect02.yaml:
    -
      from:
        line: 3
        column: 0
      to:
        line: 6
        column: 0
      content: "+\nuses:\n  voc: vocabulary1.yaml\n  customVoc: vocabulary.yaml\n\n"
    -
      from:
        line: 11
        column: 4
      to:
        line: 11
        column: 4
      content: |-
        +
        classTerm: voc.A
           
    -
      from:
        line: 18
        column: 8
      to:
        line: 18
        column: 8
      content: |-
        +
        propertyTerm: voc.text
               
    -
      from:
        line: 16
        column: 4
      to:
        line: 16
        column: 4
      content: |-
        +
        classTerm: voc.B
           
    -
      from:
        line: 20
        column: 8
      to:
        line: 20
        column: 8
      content: |-
        +
        propertyTerm: voc.newProp
               
    -
      from:
        line: 25
        column: 8
      to:
        line: 25
        column: 8
      content: |-
        +
        propertyTerm: voc.iDs
               
